extension/js/content_scripts/webmail/gmail-element-replacer.ts

```diff
@@ -103,7 +104,8 @@ export class GmailElementReplacer implements WebmailElementReplacer {
     }
   }
 
-  private everything = () => {
+  private everything = async () => {
+    this.tabId = await BrowserMsg.requiredTabId();
```

Comment:
This would be getting the tab id every second. Since it's an async call, it may be somewhat pricey to do it every second (plus it doesn't change).

I would try to instead change the signature of `await webmailSpecific.start(acctEmail, inject, notifications, factory, notifyMurdered);` to add `tabId` there as argument, then propagate it around so that you can use it inside `everything`.